memas/dataplane.py

```diff
@@ -14,21 +18,31 @@ def recall():
 
     current_app.logger.info(f"Recalling [namespace_pathname=\"{namespace_pathname}\"]")
 
+
     corpus_infos = ctx.memas_metadata.get_query_corpora(namespace_pathname)
 
     current_app.logger.debug(f"Querying corpuses: {corpus_infos}")
-    search_results: list[tuple[str, Citation]] = []
-    for corpus_info in corpus_infos:
+    # search_results: list[tuple[str, Citation]] = []
+    # for corpus_info in corpus_infos:
+    #     corpus: Corpus = ctx.corpus_provider.get_corpus_by_info(corpus_info)
+    #     search_results.extend(corpus.search(clue=clue))
+
+    # Group the corpora to search into sets based on their CorpusType
+    corpora_grouped_by_type = defaultdict(list)
+    for corpus_info in corpus_infos :
+        corpus_type = corpus_info.corpus_type
         corpus: Corpus = ctx.corpus_provider.get_corpus_by_info(corpus_info)
-        search_results.extend(corpus.search(clue=clue))
+        corpora_grouped_by_type[corpus_type].append(corpus)
 
-    # Combine the results and only take the top ones
-    search_results.sort(key=lambda x: x[0], reverse=True)
+    # Execute a multicorpus search
+    # Need to refactor to remove ctx later and have a cleaner solution, but thats time i dont have right now : (
+    search_results = mult_corpus_search(corpora_grouped_by_type, clue, ctx, 4)
+    current_app.logger.debug(f"Search Results are: {search_results}")
 
     # TODO : It will improve Query speed significantly to fetch citations after determining which documents to send to user
 
     # Take only top few scores and remove scoring element before sending
-    return [{"document": doc, "citation": asdict(citation)} for doc, citation in search_results[0:5]]
+    return [{"score" : score, "document": doc, "citation": asdict(citation)} for score, doc, citation in search_results[0:5]]
```

Comment:
Do we want to expose the scores of the search results? What benefits does it add?

Reply:
I'm also a little iffy on whether exposing scoring to users is worth doing, so I leave it up to you. 

I was thinking maybe we do want to expose the results to give developers more insight into what exactly they are retrieving and how they MAY compare with one another. When the corpora scoring isn't comparable it doesn't have obvious value, but I was thinking of how ElasticSearch queries return with scores as well (that can't necessarily be compared with other ES queries).

Reply:
Personally think let's leave it out for now, unless there's a strong demand. Main reason being it complicates the interpretation of these results, and is something we can add quite easily